src/backend/distributed/shared_library_init.c

```diff
@@ -2021,6 +2033,20 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomIntVariable(
+		"citus.max_databases_per_worker_tracked",
```

Comment:
> Tried to fix the `check_gucs_are_alphabetically_sorted` and got a bit confused: it seems to be failing on `citus.enable_repartitioned_insert_select` even on `main`, on my machine. But it somehow [passes](https://github.com/citusdata/citus/actions/runs/7731677508/job/21079902338) on your CI 🤔
> 
> ```
> ivan@ivan-pc:~/IdeaProjects/citus-fork$ ./ci/check_gucs_are_alphabetically_sorted.sh 
> sort: gucs.out:40: disorder:            "citus.enable_repartitioned_insert_select",
> ```

Looking at your changes both new gucs that you're introducing are not at the correct place alphabetically (both need to be moved up). I expect that if you fix their location in the list, that the error goes away.

I agree that it's weird that it reports `citus.enable_repartitioned_insert_select` though. Sounds like a reporting bug. @onurctirtir do you know more about this? (you changed stuff about this script recently)

Reply:
Ok, figured that out: I haven't had `LC_COLLATE=C`, so `sort` on my machine treated the underscore character differently. 
The easiest fix is to force it for the `sort` explicitly in the `check_gucs_are_alphabetically_sorted.sh`:
```
grep -P "^[\t][\t]\"citus\.[a-zA-Z_0-9]+\"" RegisterCitusConfigVariables_func_def.out > gucs.out
LC_COLLATE=C sort -c gucs.out
```

Reply:
Ah interesting, could you make a separate PR to fix that.

Reply:
Certainly :slightly_smiling_face: 
But I am not sure that the easiest fix would be the best one: may be it's better to force it via `./configure` or something like that, so every dev setup would be as identical to CI as possible

Reply:
Well, in test runs we already set collation etc to expected settings. But these are simple shell scripts where we don't have this infra structure. And normally that doesn't matter, but indeed for sorting it does.

Reply:
So, to clarify. If you open a PR where you only add LC_COLLATE=C to the sort command, then I'd be happy to accept that.

Reply:
Yes, I'll add it tomorrow, don't worry :)
Got distracted trying to reproduce the flakiness :)